internal/core/application/market_loader_service.go

```diff
@@ -62,21 +63,97 @@ func (m *marketsLoaderService) FetchMarkets() {
 		return
 	}
 
-	if err := m.marketRepository.DeleteAllMarket(context.Background()); err != nil {
-		log.Errorf("FetchMarkets -> DeleteAllMarket: %v", err)
+	//markets already stored in db
+	existingMarkets, err := m.marketRepository.GetAllMarkets(context.Background())
+	if err != nil {
+		log.Errorf("FetchMarkets -> GetAllMarkets: %v", err)
 		return
 	}
 
+	//assume that external service, from which we are fetching markets, should return active markets
+	activeMarkets := make([]domain.Market, 0)
 	for _, v := range liquidityProviders {
-		for _, v1 := range v.Markets {
-			if err := m.marketRepository.InsertMarket(context.Background(), domain.Market{
+		for _, market := range v.Markets {
+			activeMarkets = append(activeMarkets, domain.Market{
 				ProviderName: v.Name,
 				Url:          v.Endpoint,
-				BaseAsset:    v1.BaseAsset,
-				QuoteAsset:   v1.QuoteAsset,
-			}); err != nil {
-				log.Errorf("FetchMarkets -> InsertMarket: %v", err)
+				BaseAsset:    market.BaseAsset,
+				QuoteAsset:   market.QuoteAsset,
+				Active:       true,
+			})
+		}
+	}
+
+	if err := m.updateMarketActiveStatusAndInsertNew(activeMarkets, existingMarkets); err != nil {
+		log.Errorf("FetchMarkets -> updateMarketActiveStatusAndInsertNew: %v", err)
+	}
+}
+
+func (m *marketsLoaderService) updateMarketActiveStatusAndInsertNew(
+	existingMarkets []domain.Market,
+	activeMarkets []domain.Market,
+) error {
+	//since markets can go on and off, we need to update markets accordingly
+	//3 cases here:
+	//1. existing market is not in active markets -> set market as inactive
+	//2. existing market is in active markets -> set market as active
+	//3. active market is not in existing markets -> create new market
+	marketsNotInActiveList := make([]domain.Market, 0) //to be inactivated
+	marketsInActiveList := make([]domain.Market, 0)    //to be activated
+	for _, v1 := range existingMarkets {
+		found := false
+		for _, v2 := range activeMarkets {
+			if isSameMarket(v1, v2) {
+				found = true
+				marketsInActiveList = append(marketsInActiveList, v1)
+			}
+		}
+
+		if !found {
+			marketsNotInActiveList = append(marketsNotInActiveList, v1)
+		}
+	}
```

Comment:
To improve the performances here, it would be better to have a map of the active markets. This way, we don't need to loop over a slice to know whether an existing market is still active.